core/src/trezor/ui/layouts/tr/__init__.py

```diff
@@ -1066,21 +1080,46 @@ async def confirm_modify_output(
     amount_change: str,
     amount_new: str,
 ) -> None:
-    await raise_if_not_confirmed(
-        interact(
-            RustLayout(
-                trezorui2.confirm_modify_output(
-                    address=address,
-                    sign=sign,
-                    amount_change=amount_change,
-                    amount_new=amount_new,
-                )
-            ),
-            "modify_output",
-            ButtonRequestType.ConfirmOutput,
+    address_layout = RustLayout(
+        trezorui2.confirm_blob(
+            title="MODIFY AMOUNT",
+            data=address,
+            verb="CONTINUE",
+            verb_cancel=None,
+            description="Address:",
+            extra=None,
+        )
+    )
+    modify_layout = RustLayout(
+        trezorui2.confirm_modify_output(
+            sign=sign,
+            amount_change=amount_change,
+            amount_new=amount_new,
         )
     )
 
+    send_button_request = True
+    while True:
+        if send_button_request:
+            await button_request(
+                "modify_output",
+                ButtonRequestType.ConfirmOutput,
+                address_layout.page_count(),
+            )
+        await raise_if_not_confirmed(ctx_wait(address_layout))
+
+        if send_button_request:
+            send_button_request = False
+            await button_request(
+                "modify_output",
+                ButtonRequestType.ConfirmOutput,
+                modify_layout.page_count(),
+            )
+        result = await ctx_wait(modify_layout)
```

Comment:
I'm afraid you'll have to call `.request_complete_repaint()` on the two layouts otherwise they'll paint over each other.

Reply:
Is this situation different in `TR` than in `TT`? This is the same exact code as in `confirm_modify_output`'s code in `core/src/trezor/ui/layouts/tt/__init__.py`

Reply:
Seems to be the same & in both cases there is garbage on screen when you go back from the MODIFY AMOUNT screen in `test_p2wpkh_in_p2sh_fee_bump_from_external`.

TBH this should be handled automagically somewhere in `class Layout`, not sure if it's worth spending time on fixing it or rewriting to rust ...

Reply:
Good catch. I did the quick fix in [4970a6b](https://github.com/trezor/trezor-firmware/pull/3393/commits/4970a6bdce3c90b58cef13a3edca60dc9c643dfa) --- requesting repaint in all `while True:` cases where we define the layout in front of the loop